jlc_kicad_tools/jlc_lib/cpl_fix_rotations.py

```diff
@@ -68,11 +74,19 @@ def FixRotations(input_filename, output_filename, db):
           if row[i] in HEADER_REPLACEMENT_TABLE:
             row[i] = HEADER_REPLACEMENT_TABLE[row[i]]
       else:
+        rotation = float(row[rotation_index])
+        if posx_index is not None and side_index is not None:
+            if row[side_index].strip() == "bottom":
+                row[posx_index] = "{0:.6f}".format(-float(row[posx_index]))
         for pattern, correction in db.items():
           if pattern.match(row[package_index]):
             _LOGGER.logger.info("Footprint {} matched {}. Applying {} deg correction"
                 .format(row[package_index], pattern.pattern, correction))
-            row[rotation_index] = "{0:.6f}".format((float(row[rotation_index]) + correction) % 360)
+            if side_index is not None and row[side_index].strip() == "bottom":
```

Comment:
Can you add checks for posx_index and side_index up above instead (after the "if rotation_index is None" check)?

If we can't find any of these fields (likely because kicad changed the format) the output of the script will probably be wrong, and it would be good to die with an informative error message in that case, instead of silently ignoring it.

Reply:
Agreed, done!